### DISCUSSION TITLE
Introduce Pipeline (|>) Operator

### DISCUSSION THREAD

**Discussion by leandromoh** [dotnet/csharplang#4599] [Language Ideas] [answered]
Introduces a new operator `|>` similar to [F#](https://en.wikibooks.org/wiki/F_Sharp_Programming/Higher_Order_Functions#The_.7C.3E_Operator), [OCaml](http://caml.inria.fr/pub/docs/manual-ocaml/libref/Pervasives.html#VAL%28|%3E%29), [Elixir](https://hexdocs.pm/elixir/Kernel.html#%7C%3E/2), [Elm](https://package.elm-lang.org/packages/elm/core/latest/Basics#|%3E), [Julia](https://docs.julialang.org/en/v1/base/base/#Base.:|%3E), [Hack](https://docs.hhvm.com/hack/expressions-and-operators/pipe), [Clojure](https://clojure.github.io/clojure/clojure.core-api.html#clojure.core/as->), and [LiveScript](http://livescript.net/#piping).

`64 |> sqrt`  is equivalent to `sqrt(64)`

This allows for greater readability when chaining several functions together. For example, given the following functions:
```c#
string DoubleSay (string str) {
  return str + ", " + str;
}

string Capitalize (string str) {
  return char.ToUpper(str[0]) + str.Substring(1).ToLower();
}

string Exclaim (string str) {
  return str + '!';
}
```
...the following invocations are equivalent:
```c#
var result = Exclaim(Capitalize(DoubleSay("hello")));
result //=> "Hello, hello!"

var result = "hello"
  |> DoubleSay
  |> Capitalize
  |> Exclaim;

result //=> "Hello, hello!"
```

multiple argument functions
```c#
let newScore = "70"
  |> int.Parse
  |> (_ =>  3 + _ )
  |> (_ => BoundScore(0, 100, _));

// As opposed to: let newScore = BoundScore( 0, 100, int.Parse("70") + 3 )
```
underscore is not required; it's just a lambda function, so you can use any parameter name you like.

**Comment by HaloFour** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/discussions/74
